core/capabilities/ccip/ccipevm/msghasher_test.go

```diff
@@ -349,6 +354,97 @@ func TestMessagerHasher_againstRmnSharedVector(t *testing.T) {
 		require.Equal(t, rmnMsgHash, msgH.String(), "rmn hash and my hash should match")
 	})
 
+	t.Run("solana vector", func(t *testing.T) {
+		key, err := solanago.NewRandomPrivateKey()
+		require.NoError(t, err)
+
+		// Borsh encoded extra args
+		ea := fee_quoter.EVMExtraArgsV2{
+			GasLimit:                 agbinary.Uint128{Lo: 5000, Hi: 0},
+			AllowOutOfOrderExecution: false,
+		}
+
+		var extraArgsbuf bytes.Buffer
+		encoder := agbinary.NewBorshEncoder(&extraArgsbuf)
+		err = ea.MarshalWithEncoder(encoder)
+		require.NoError(t, err)
+
+		destGasAmount := uint32(10)
+		destExecData := make([]byte, 4)
+		binary.LittleEndian.PutUint32(destExecData, destGasAmount)
+
+		var (
+			// header fields
+			messageID           = mustBytes32FromString(t, "0xcdad95e113e35cf691295c1f42455d41062ba9a1b96a6280c1a5a678ef801721")
+			sourceChainSelector = cciptypes.ChainSelector(124615329519749607)   // solana
+			destChainSelector   = cciptypes.ChainSelector(16015286601757825753) // sepolia
+			sequenceNumber      = cciptypes.SeqNum(386)
+			nonce               = uint64(1)
+			// message fields
+			// sender is parsed unpadded since its emitted unpadded from EVM.
```

Comment:
```suggestion
			// sender is parsed unpadded since its emitted unpadded from SVM.
```

Reply:
Maybe I should just remove this comment ?

Reply:
These kinda details are critical for EVM because we need to know if it has to get padded or not. Not sure whats the case for SVM.